common/src/main/java/org/apache/drill/common/scanner/RunTimeScan.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.common.scanner;
+
+import static java.util.Collections.emptySet;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.apache.drill.common.config.DrillConfig;
+import org.apache.drill.common.scanner.persistence.ScanResult;
+
+/**
+ * Utility to scan classpath at runtime
+ *
+ */
+public class RunTimeScan {
+
+  // result of prescan
+  private static final ScanResult PRESCANNED = BuildTimeScan.load();
+
+  // urls of the locations (classes directory or jar) to scan that don't have a registry in them
+  private static final Collection<URL> NON_PRESCANNED_MARKED_PATHS = getNonPrescannedMarkedPaths();
+
+  // one element cache
+  private static Set<String> SCANNED_PACKAGE_PREFIXES_FROM_CONFIG = emptySet();
```

Comment:
Since you're modifying interfaces anyway, can we please remove this static cache? At a minimum, I'd rather carry it on the DrillConfig object than in a static.